[patch] Implement core file's PID for s390* and ppc*
http://sourceware.org/ml/binutils/2011-04/msg00272.html
http://sourceware.org/ml/binutils-cvs/2011-05/msg00056.html

### src/bfd/ChangeLog	2011/05/09 13:23:24	1.5338
### src/bfd/ChangeLog	2011/05/10 06:13:05	1.5339
## -1,3 +1,10 @@
+2011-05-10  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
+	* elf.c (elfcore_grok_psinfo): Initialize CORE_PID for both native and
+	32bit psinfo.
+	* elf32-ppc.c (ppc_elf_grok_psinfo): Initialize core_pid.
+	* elf64-ppc.c (ppc64_elf_grok_psinfo): Likewise.
+
 2011-05-09  Paul Brook  <paul@codesourcery.com>
 
 	* bfd-in.h (elf32_tic6x_fix_exidx_coverage): Add prototype.
Index: gdb-7.2/bfd/elf.c
===================================================================
--- gdb-7.2.orig/bfd/elf.c	2011-08-10 21:04:07.000000000 +0200
+++ gdb-7.2/bfd/elf.c	2011-08-10 21:30:17.000000000 +0200
@@ -7818,14 +7818,14 @@ elfcore_grok_psinfo (bfd *abfd, Elf_Inte
       elf_tdata (abfd)->core_command
 	= _bfd_elfcore_strndup (abfd, psinfo.pr_psargs,
 				sizeof (psinfo.pr_psargs));
+      elf_tdata (abfd)->core_pid = psinfo.pr_pid;
     }
 #if defined (HAVE_PRPSINFO32_T) || defined (HAVE_PSINFO32_T)
   else if (note->descsz == sizeof (elfcore_psinfo32_t))
-    {
-      /* 64-bit host, 32-bit corefile */
+    { /* 64-bit host, 32-bit corefile */
       elfcore_psinfo32_t psinfo;
-
       memcpy (&psinfo, note->descdata, sizeof (psinfo));
+      elf_tdata (abfd)->core_pid = psinfo.pr_pid;
 
       elf_tdata (abfd)->core_program
 	= _bfd_elfcore_strndup (abfd, psinfo.pr_fname,
Index: gdb-7.2/bfd/elf32-ppc.c
===================================================================
--- gdb-7.2.orig/bfd/elf32-ppc.c	2011-08-10 21:04:07.000000000 +0200
+++ gdb-7.2/bfd/elf32-ppc.c	2011-08-10 21:04:09.000000000 +0200
@@ -1881,6 +1881,8 @@ ppc_elf_grok_psinfo (bfd *abfd, Elf_Inte
       return FALSE;
 
     case 128:		/* Linux/PPC elf_prpsinfo.  */
+      elf_tdata (abfd)->core_pid
+	= bfd_get_32 (abfd, note->descdata + 16);
       elf_tdata (abfd)->core_program
 	= _bfd_elfcore_strndup (abfd, note->descdata + 32, 16);
       elf_tdata (abfd)->core_command
Index: gdb-7.2/bfd/elf64-ppc.c
===================================================================
--- gdb-7.2.orig/bfd/elf64-ppc.c	2011-08-10 21:04:07.000000000 +0200
+++ gdb-7.2/bfd/elf64-ppc.c	2011-08-10 21:04:09.000000000 +0200
@@ -2668,6 +2668,8 @@ ppc64_elf_grok_psinfo (bfd *abfd, Elf_In
   if (note->descsz != 136)
     return FALSE;
 
+  elf_tdata (abfd)->core_pid
+    = bfd_get_32 (abfd, note->descdata + 24);
   elf_tdata (abfd)->core_program
     = _bfd_elfcore_strndup (abfd, note->descdata + 40, 16);
   elf_tdata (abfd)->core_command
